internal/util/kubernetes/unstructured_util.go

```diff
@@ -51,6 +51,26 @@ func ParseStatus(obj *GenericObject) (GenericStatus, error) {
 	return status, nil
 }
 
+// TODO: This is a temporary function which will be removed once all the controller are migrated to use Unstructured Object
+func ParseStatusUnstructured(obj *unstructured.Unstructured) (GenericStatus, error) {
+	statusRaw, found, err := unstructured.NestedFieldNoCopy(obj.Object, "status")
+	if !found || err != nil {
+		return GenericStatus{}, nil
+	}
+
+	var status GenericStatus
+	objByte, err := json.Marshal(statusRaw)
+	if err != nil {
+		return GenericStatus{}, err
+	}
+	err = json.Unmarshal(objByte, &status)
+	if err != nil {
+		return GenericStatus{}, err
+	}
```

Comment:
nit: can you use the [util.StructToStruct() ](https://github.com/numaproj/numaplane/blob/main/internal/util/util.go#L11)method here?

Reply:
I'm okay if you want to merge this PR and do this as a follow up

Reply:
Actually, in the interest of moving forward, I think I'll just merge this so I can merge main into my PR and resolve the conflicts.